File: snakecode/lib/linkedlist.py
from typing import List

class ListNode:
    ''' Node for singly-linked list '''
    def __init__(self, val: int = 0, next = None):
        self.val = val
        self.next = next

    def __str__(self):
        return str(listize(self))
        
def linkerize(nums: List) -> ListNode:
    ''' Convert Python list to singly-linked list '''
    fake = ListNode()
    pre = fake
    for v in nums:
        pre.next = ListNode(v)
        pre = pre.next
    return fake.next

def listize(head: ListNode) -> List:
    ''' Convert singly-linked list to Python list '''
    result = []
    while head:
        result.append(head.val)
        head = head.next
    return result

File: gocode/util/linkedlist.go
package util

// ListNode is the node of a linkedlist
type ListNode struct {
	Val  int
	Next *ListNode
}

// Linkerize converts a slice to a linkedlist, return the head of the linkedlist
func Linkerize(nums []int) *ListNode {
	fake := &ListNode{}
	cur := fake
	for _, v := range nums {
		(*cur).Next = &ListNode{v, nil}
		cur = (*cur).Next
	}
	return fake.Next
}

// Listize convert a linkedlist to a slice, using the head of the linkedlist
func Listize(node *ListNode) []int {
	result := []int{}
	for node != nil {
		result = append(result, node.Val)
		node = node.Next
	}
	return result
}
